apps/standalone/src/app/components/Login/LoginPage.tsx

```diff
@@ -1,190 +1,166 @@
 
-const nowInSeconds = () => Math.floor(Date.now() / 1000);
+  if (!response.ok) {
+    throw new Error(`Login redirect failed with status ${response.status}`);
+  }
 
-// Simple JWT format validation - checks if token has 3 parts separated by dots
-export const isValidJwtTokenFormat = (token: string): boolean => {
-  if (!token) return false;
-  const parts = token.split('.');
-  if (parts.length !== 3) return false;
-  // Check that each part contains only valid base64url characters
-  const base64urlPattern = /^[A-Za-z0-9_-]+$/;
-  return parts.every((part) => part.length > 0 && base64urlPattern.test(part));
+  const { url } = (await response.json()) as { url?: string };
+  if (!url) {
+    throw new Error('Login redirect URL missing in response');
+  }
+  window.location.href = url;
 };
 
-export const LoginPage = () => {
+const LoginPage = () => {
   const { t } = useTranslation();
-  const [token, setToken] = React.useState('');
-  const [validationError, setValidationError] = React.useState<string>('');
-  const [isSubmitting, setIsSubmitting] = React.useState(false);
-  const [submitError, setSubmitError] = React.useState<string>();
-
-  const handleSubmit = async (e: React.FormEvent) => {
-    e.preventDefault();
-    setSubmitError(undefined);
-    setIsSubmitting(true);
-
-    try {
-      localStorage.removeItem(EXPIRATION);
-      localStorage.removeItem(ORGANIZATION_STORAGE_KEY);
-
-      const resp = await fetch(loginAPI, {
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        credentials: 'include',
-        method: 'POST',
-        body: JSON.stringify({
-          token: token.trim(),
-        }),
-      });
-
-      if (!resp.ok) {
-        let errorMessage = t('Authentication failed');
-        try {
-          const contentType = resp.headers.get('content-type');
-          if (contentType && contentType.includes('application/json')) {
-            const errorData = (await resp.json()) as { error?: string };
-            errorMessage = errorData.error || errorMessage;
-          } else {
-            // Fallback for non-JSON responses
-            const text = await resp.text();
-            if (text) {
-              errorMessage = text;
+  const { get } = useFetch();
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState<string>();
+  const [providers, setProviders] = React.useState<AuthProvider[]>([]);
+  const [userSelectedProvider, setUserSelectedProvider] = React.useState<AuthProvider | null>(null);
+  const [defaultProviderName, setDefaultProviderName] = React.useState<string>('');
+  const [isRedirecting, setIsRedirecting] = React.useState(false);
+
+  const handleProviderSelect = async (provider: AuthProvider) => {
+    // Prevent multiple clicks while redirect is in progress
+    if (isRedirecting) {
+      return;
+    }
+
+    setUserSelectedProvider(provider);
```

Comment:
When user tries again, we should remove the previous error
```suggestion
    setError(undefined)
    setUserSelectedProvider(provider);
```

Reply:
In general, the error handling did not work well if prefferedUsername was not present. Not a blocker though.